### PR TITLE
Include missing attributes when upgrading embed block

### LINKED ISSUE #33231: Attributes are lost when changing the provider of an embed block
## Description

The following attributes of the embed block are lost when the provider is changed:
- `align`
- `caption`
- `allowResponsive`

## Step-by-step reproduction instructions

1. Add an embed block.
2. Set an URL (YouTube video example: https://youtu.be/UOcroR50808).
3. Click on `Embed` button.
4. Change alignment to `Align left`.
5. Add text to the caption field.
6. Click on `Show more settings` (if they're not already displayed).
7. Uncheck `Resize for smaller devices` option.
8. Edit the URL by clicking on the ✏️ button located in the block's toolbar.
9. Set a URL from a different provider (Spotify song example: https://open.spotify.com/track/6UB9mShVLbMm0W4e6vud4C?si=3b03fa64857f4919).
10. Click on `Embed` button.
11. Observe that alignment changed to the default value (`Align center`).
12. Observe that the caption text is empty.
13. Observe that `Resize for smaller devices` option is now checked (the default value).

## Expected behaviour

Attributes should be preserved when the URL and provider change.

## Actual behaviour

Attributes are not preserved when the URL and provider change.

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/14905380/124641368-eabe6280-de8e-11eb-92ec-e5e3f12285df.mp4

## WordPress information
- WordPress version: 5.8-alpha-50771
- Gutenberg version: 11.0.0-rc.1
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? Yes

## Device information
- Device: Desktop
- Operating system: MacOS 10.15.7
- Browser: Chrome 91.0.4472.114

### PR DESCRIPTION
**Pull request by fluiddot**
## Description

Include the following attributes that were missing when an embed block is upgraded due to finding a more suitable block:
- `align`
- `caption`
- `allowResponsive`

## How has this been tested?

1. Add an embed block.
2. Set an URL (YouTube video example: https://youtu.be/UOcroR50808).
3. Click on `Embed` button.
4. Change alignment to `Align left`.
5. Add text to the caption field.
6. Click on `Show more settings` (if they're not already displayed).
7. Uncheck `Resize for smaller devices` option.
8. Edit the URL by clicking on the ✏️ button located in the block's toolbar.
9. Set a URL from a different provider (Spotify song example: https://open.spotify.com/track/6UB9mShVLbMm0W4e6vud4C?si=3b03fa64857f4919).
10. Click on `Embed` button.
11. Observe that alignment is preserved (the value should be `Align left`).
12. Observe that the caption is preserved (the value should be the text set previously).
13. Observe that `Resize for smaller devices` option is preserved (the option should be unchecked).

## Screenshots 

https://user-images.githubusercontent.com/14905380/124643637-9b2d6600-de91-11eb-8c16-2d848d3c5cf2.mp4

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by Mamaduka — APPROVED** [review]
Tested and attributes are preserved when changing the embed source.

LGTM 🚢

**Comment by fluiddot** [comment]
> Tested and attributes are preserved when changing the embed source.
> 
> LGTM 🚢

Thanks for the review @Mamaduka! I'll update the `block-library` changelog to include this change and merge it.

**Comment by fluiddot** [comment]
> > Tested and attributes are preserved when changing the embed source.
> > LGTM 🚢
> 
> Thanks for the review @Mamaduka! I'll update the `block-library` changelog to include this change and merge it.

`block-library` changelog updated in [this commit](https://github.com/WordPress/gutenberg/pull/33235/commits/3709b6657b7977b37047aacdbb5cb400ac4be83f).

File: packages/block-library/src/embed/util.js [hunk 1]
```diff
@@ -114,6 +124,9 @@ export const createUpgradedEmbedBlock = (
 			! providerNameSlug );
 	if ( shouldCreateNewBlock ) {
 		return createBlock( DEFAULT_EMBED_BLOCK, {
+			align,
```
Comment (ntsekouras) [inline packages/block-library/src/embed/util.js]: Sorry for being late here, but doesn't it make sense to just add all attributes here?

Something like that:
```
const { preview, attributes = {} } = props;
const { url, providerNameSlug, type, ...restAttributes } = attributes;
....
return createBlock( DEFAULT_EMBED_BLOCK, {
	url,
	...restAttributes,
	...matchedBlock.attributes,
} );
```

Comment (fluiddot) [reply packages/block-library/src/embed/util.js]: Yeah, I think we could include all of them, my initial concern was if the extra ones would produce any kind of side-effect:
- `className`
- `previewable`
- `responsive`

So far what I tested looks safe so I'm going to apply your suggestion, thanks @ntsekouras 🙇 !

Comment (fluiddot) [reply packages/block-library/src/embed/util.js]: > So far what I tested looks safe so I'm going to apply your suggestion, thanks @ntsekouras 🙇 !

I added this suggestion in [this commit](https://github.com/WordPress/gutenberg/pull/33235/commits/f9c347f369761c7b084e95dc5a18786f09a0bce4).

**Review by ntsekouras — APPROVED** [review]
Looks good! Thank you for working on this! 👍

**Pull request merged**